src/AutoRest.CSharp/Common/Output/Models/Types/ObjectType.cs

```diff
@@ -3,107 +3,72 @@
-
-            var hasUsage = _usage.HasFlag(SchemaTypeUsage.Model);
-
-            DefaultAccessibility = objectSchema.Extensions?.Accessibility ?? (hasUsage ? "public" : "internal");
-            Description = BuilderHelpers.CreateDescription(objectSchema);
-
-            DefaultNamespace = GetDefaultNamespace(objectSchema, context);
-            _sourceTypeMapping = context.SourceInputModel?.CreateForModel(ExistingType);
-
-            // Update usage from code attribute
-            if (_sourceTypeMapping?.Usage != null)
-            {
-                foreach (var usage in _sourceTypeMapping.Usage)
-                {
-                    _usage |= Enum.Parse<SchemaTypeUsage>(usage, true);
-                }
-            }
         }
 
-        protected ObjectSchema OjectSchema { get; }
-
-        public bool IsStruct => ExistingType?.IsValueType == true;
-        protected override string DefaultName => OjectSchema.CSharpName();
-        protected override string DefaultAccessibility { get; } = "public";
-        protected override string DefaultNamespace { get; }
-        protected override TypeKind TypeKind => IsStruct ? TypeKind.Struct : TypeKind.Class;
-
-        public string? Description { get; }
+        public ObjectTypeConstructor[] Constructors => _constructors ??= BuildConstructors().ToArray();
+        public virtual ObjectTypeProperty[] Properties => _properties ??= BuildProperties().ToArray();
 
         public virtual CSharpType? Inherits => _inheritsType ??= CreateInheritedType();
-
+        public ObjectTypeConstructor? SerializationConstructor => _serializationConstructor ??= BuildSerializationConstructor();
+        public CSharpType? ImplementsDictionaryType => _implementsDictionaryType ??= CreateInheritedDictionaryType();
+        public ObjectTypeConstructor InitializationConstructor => _initializationConstructor ??= BuildInitializationConstructor();
         public ObjectSerialization[] Serializations => _serializations ??= BuildSerializations();
+        public bool IsStruct => ExistingType?.IsValueType == true;
+        public virtual string? Description { get; protected set; }
+        public ObjectTypeDiscriminator? Discriminator => _discriminator ??= BuildDiscriminator();
 
-        public ObjectTypeProperty[] Properties => _properties ??= BuildProperties().ToArray();
+        public abstract bool IncludeSerializer { get; }
+        public abstract bool IncludeDeserializer { get; }
+        public abstract bool IncludeConverter { get; }
+        public abstract ObjectTypeProperty? AdditionalPropertiesProperty { get; }
+        protected abstract ObjectTypeDiscriminator? BuildDiscriminator();
+        protected abstract ObjectSerialization[] BuildSerializations();
+        protected abstract ObjectTypeConstructor BuildInitializationConstructor();
+        protected abstract CSharpType? CreateInheritedDictionaryType();
+        protected abstract ObjectTypeConstructor? BuildSerializationConstructor();
```

Comment:
I'm not a fan of having the mixed extensibility story where some properties are virtual by themselves and other have virtual caching builders. I would pick one.

Reply:
not sure I follow

Reply:
To restate, why do we need both these:

`public ObjectTypeConstructor SerializationConstructor => _serializationConstructor ??= BuildSerializationConstructor();`
And
`protected abstract ObjectTypeConstructor BuildSerializationConstructor();`

and then later have

`public abstract bool IncludeDeserializer { get; }`

Some things are properties you need to define values from, others are methods that the base class uses to define properties for.

If we made `ObjectTypeConstructor SerializationConstructor { get; }` abstract then it would be consistent is the point I believe.

Reply:
that is how it was before.  I am not introducing this its a copy paste from the old objecttype